control/metrics.go

```diff
@@ -55,17 +55,28 @@ var (
 
 // hostnameReader, hostnamer created for mocking
 func init() {
-	hostnameReader = &hostnameReaderType{}
+	h, err := os.Hostname()
+	if err != nil {
+		log.WithFields(log.Fields{
+			"_module": "control",
+			"_file":   "metrics.go,",
+			"_block":  "addStandardAndWorkflowTags",
+			"error":   err.Error(),
+		}).Error("Unable to determine hostname")
+	}
+	hostnameReader = &hostnameReaderType{hostname: h}
```

Comment:
Just that it was the intention of the PR. Is there a case where the hostname would change more regularly than snapd would be restarting?